tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/VectorFloat2.java

```diff
@@ -0,0 +1,225 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * GNU Classpath is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ * 
+ * GNU Classpath is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with GNU Classpath; see the file COPYING.  If not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * Linking this library statically or dynamically with other modules is
+ * making a combined work based on this library.  Thus, the terms and
+ * conditions of the GNU General Public License cover the whole
+ * combination.
+ * 
+ * As a special exception, the copyright holders of this library give you
+ * permission to link this library with independent modules to produce an
+ * executable, regardless of the license terms of these independent
+ * modules, and to copy and distribute the resulting executable under
+ * terms of your choice, provided that you also meet, for each linked
+ * independent module, the terms and conditions of the license of that
+ * module.  An independent module is a module which is not derived from
+ * or based on this library.  If you modify this library, you may extend
+ * this exception to your version of the library, but you are not
+ * obligated to do so.  If you do not wish to do so, delete this
+ * exception statement from your version.
+ *
+ */
+package uk.ac.manchester.tornado.api.collections.types;
+
+import static java.lang.String.format;
+import static java.nio.FloatBuffer.wrap;
+import static uk.ac.manchester.tornado.api.collections.types.Float2.add;
+import static uk.ac.manchester.tornado.api.collections.types.Float2.loadFromArray;
+import static uk.ac.manchester.tornado.api.collections.types.FloatOps.fmt3;
+
+import java.nio.FloatBuffer;
+
+public class VectorFloat2 implements PrimitiveStorage<FloatBuffer> {
+
+    /**
+     * backing array
+     */
+    final protected float[] storage;
+
+    /**
+     * number of elements in the storage
+     */
+    final private int numElements;
+    final private static int elementSize = 2;
+
+    /**
+     * Creates a vector using the provided backing array
+     *
+     * @param numElements
+     * @param array
+     */
+    protected VectorFloat2(int numElements, float[] array) {
+        this.numElements = numElements;
+        this.storage = array;
+    }
+
+    /**
+     * Creates a vector using the provided backing array
+     */
+    public VectorFloat2(float[] array) {
+        this(array.length / elementSize, array);
```

Comment:
Remove this division. Keep consistency from the rest of the API.

Reply:
The original implementation of the already supported`VectorFloat3` and `VectorFloat4` use this allocation strategy, so I kept consistency with these two. 
I think is division is there, because originally it was meant to pack the inputs into vectors with primitive types of lenghX rather than create Vector types matching the input size.